### ISSUE TITLE
Some questions in step function

### ISSUE THREAD

**Issue by RookieXwc** [sail-sg/Adan#17]
Thank you for your impressive work. I have some questions in your `adan.py` about `step` function.
In line 179-180, that is:
``` 
for p, copy_grad in zip(group['params'], copy_grads):
    self.state[p]['pre_grad'] = copy_grad
```
It seems that you want to save the corresponding `pre_grad`. But I have the following bug:
![image](https://user-images.githubusercontent.com/44500103/198328597-c04d345b-5a96-4288-992c-4a7510796d1c.png)
I think this is because the former contains all parameters, while the latter only contains parameters with gradient. So I made the following changes:
``` 
for p, copy_grad in zip(params_with_grad, copy_grads):
    self.state[p]['pre_grad'] = copy_grad
```
With this modification, I can run normally. Do you think what problems I have encountered and that this modification is correct? @XingyuXie

**Comment by XingyuXie** [comment]
@RookieXwc   Thanks a lot you are right. 

It seems that some of your parameters are set to `requires_grad=False`, right? Since I do not meet this problem, but your modification is right.

**Comment by RookieXwc** [comment]
Yes, some of my parameters are set to `requires_grad=False`. Therefore, the lengths of `group['params']` and `copy_grads` are inconsistent. I noticed that this error will not occur in the `timm` version. Could you clearly check in your repo and make corresponding modifications? Thank you!

**Comment by XingyuXie** [comment]
Done, thanks again @RookieXwc .

BZW, the newer version is faster than the older or timm version, especially on a single node.
